tests/test_adapters.py

```diff
@@ -57,66 +75,145 @@ async def test_prompt_adapter():
 
 
 @pytest.mark.asyncio
-async def test_lora_adapter():
+async def test_lora_adapter(vllm_model_handler):
+    base_model_name = "/granite/granite-3b-base-v2/step_75000_ckpt"
     adapter_name = "granite-3b-code-instruct-lora"
+    lora_path = str(FIXTURES_DIR) + "/" + adapter_name
     request = BatchedGenerationRequest(
         adapter_id=adapter_name,
+        model_id=base_model_name,
     )
+    mock_lora_requests = PropertyMock(
+        side_effect=[
+            [],
+            [
+                LoRARequest(
+                    lora_name=adapter_name,
+                    lora_int_id=0,
+                    lora_path=lora_path,
+                ),
+            ],
+        ]
+    )
+    type(vllm_model_handler).lora_requests = mock_lora_requests
```

Comment:
`type(vllm_model_handler)` should resolve to the class `AsyncMock` so I think this will set `.lora_requests` for every instance of an AsyncMock from here on out...? Not sure that's quite what we want.

Is the issue that when you do `vllm_model_handler.lora_requests` it expects that to be an awaitable?

Reply:
i'm not 100% clear on what the mock is doing here, but when i just assign `vllm_model_handler.lora_requests" to the PropertyMock, I have to call `vllm_model_handler.lora_requests()` to get the expected value out of it, when I want to reference it as an attribute `vllm_model_handler.lora_requests`.

Reply:
I looked into it a bit more, and it looks to me like the `type(vllm_model_handler)` behavior actually returns a different instance of a class for each mock, e.g.
```
print( type(Mock()) == type(Mock()) )
```
prints `False`. So I think this actually is safe to do, it's just... pretty interesting. But I guess that's how it goes for python mocks